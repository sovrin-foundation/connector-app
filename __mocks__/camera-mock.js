// @flow
import React from 'react'
import { View } from 'react-native'

const constants = {
  Aspect: {},
  BarCodeType: {},
  Type: {},
  CaptureMode: {},
  CaptureTarget: {},
  CaptureQuality: {},
  Orientation: {},
  FlashMode: {},
  TorchMode: {},
}

// We don't need to type check external library
// and we don't need types for mock test stub
class MockCamera extends React.Component<any, void> {
  static constants = constants
  static checkVideoAuthorizationStatus = jest.fn(() => Promise.resolve(true))

  render() {
    return <View>{this.props.children}</View>
  }
}

MockCamera.constants = constants

export default MockCamera
