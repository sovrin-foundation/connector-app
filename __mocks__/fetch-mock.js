import { Headers, Response, Request } from 'node-fetch'

const fetch = jest.fn(() => {})
fetch.Headers = Headers
fetch.Request = Request
fetch.Response = Response

fetch.mockResponse = (body, init) =>
  fetch.mockImplementation(
    () =>
      new Promise((resolve, reject) =>
        process.nextTick(() => resolve(new Response(body, init)))
      )
  )

fetch.mockResponseOnce = (body, init) =>
  fetch.mockImplementationOnce(
    () =>
      new Promise((resolve, reject) =>
        process.nextTick(() => resolve(new Response(body, init)))
      )
  )

fetch.mockResponses = (...responses) =>
  responses.map(([body, init]) =>
    fetch.mockImplementationOnce(
      () =>
        new Promise((resolve, reject) =>
          process.nextTick(() => resolve(new Response(body, init)))
        )
    )
  )

export default fetch
