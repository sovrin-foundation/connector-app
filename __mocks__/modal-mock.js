// @flow
import React from 'react'

const MockModal = (props: any) =>
  React.createElement('Modal', props, props.children)

export default MockModal
