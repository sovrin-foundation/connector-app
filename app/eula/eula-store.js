// @flow
import { takeLatest, call, put, all } from 'redux-saga/effects'
import { safeSet } from '../services/storage'
import {
  EULA_ACCEPT,
  STORAGE_KEY_EULA_ACCEPTANCE,
  HYDRATE_EULA_ACCEPT,
} from './type-eula'
import type {
  EulaAccept,
  EulaStore,
  EulaActions,
  HydrateEulaAcceptAction,
} from './type-eula'
import { captureError } from '../services/error/error-handler'

const initialState: EulaStore = {
  isEulaAccept: false,
}

export const eulaAccept = (isEulaAccept: boolean): EulaAccept => ({
  type: EULA_ACCEPT,
  isEulaAccept,
})

// if we see that both this action and eulaAccept action creator
// does the same thing inside the reducer and while raising action as well
// the reason we have two action creators is that on EULA_ACCEPT
// we have a watch that runs and store this data inside wallet
// we do not want to wastefully make an API call when we just hydrated data
export const hydrateEulaAccept = (
  isEulaAccept: boolean
): HydrateEulaAcceptAction => ({
  type: HYDRATE_EULA_ACCEPT,
  isEulaAccept,
})

export function* watchEula(): any {
  yield all([watchEulaAcceptance()])
}

export function* watchEulaAcceptance(): any {
  yield takeLatest(EULA_ACCEPT, eulaAcceptanceSaga)
}

export function* eulaAcceptanceSaga(action: EulaAccept): Generator<*, *, *> {
  try {
    const { isEulaAccept } = action

    yield call(
      safeSet,
      STORAGE_KEY_EULA_ACCEPTANCE,
      JSON.stringify(isEulaAccept)
    )
  } catch (e) {
    captureError(e)
    console.error(`eulaAcceptanceSaga: ${e}`)
  }
}

export default function eulaReducer(
  state: EulaStore = initialState,
  action: EulaActions
): EulaStore {
  switch (action.type) {
    case HYDRATE_EULA_ACCEPT:
    case EULA_ACCEPT:
      return {
        ...state,
        isEulaAccept: action.isEulaAccept,
      }
    default:
      return state
  }
}
