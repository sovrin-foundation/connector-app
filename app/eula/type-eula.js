// @flow
import type { CustomError } from '../common/type-common'
import { Platform } from 'react-native'

const isAndroid = Platform.OS === 'android'
export const EULA_ACCEPT = 'EULA_ACCEPT'
export const ANDROID_EULA_URL = 'https://www.connect.me/google.html'
export const IOS_EULA_URL = 'https://www.connect.me/ios_eula.html'
export const EULA_URL = isAndroid ? ANDROID_EULA_URL : IOS_EULA_URL
export const localEulaSource = isAndroid
  ? 'file:///android_asset/external/connectme/eula_android.html'
  : './eula_ios.html'
export const STORAGE_KEY_EULA_ACCEPTANCE = 'STORAGE_KEY_EULA_ACCEPTANCE'
export const HYDRATE_EULA_ACCEPT = 'HYDRATE_EULA_ACCEPT'

export type EulaStore = {
  isEulaAccept: boolean,
}

export type EulaActions = EulaAccept | HydrateEulaAcceptAction

export type EulaAccept = {
  type: typeof EULA_ACCEPT,
  isEulaAccept: boolean,
}

export type HydrateEulaAcceptAction = {
  type: typeof HYDRATE_EULA_ACCEPT,
  isEulaAccept: boolean,
}

export type EulaScreenState = {
  error: null | CustomError,
}
