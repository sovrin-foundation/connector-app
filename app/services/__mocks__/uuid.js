// @flow
import { defaultUUID } from '../../../__mocks__/static-data'

export function uuid() {
  return defaultUUID
}
