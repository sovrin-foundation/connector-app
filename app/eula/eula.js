// @flow

// this is page displays, a webview with of our terms and conditions
// you should get the url value and title from constants
// on click accept take user to lock selection screen

import React, { PureComponent } from 'react'
import { WebView, Alert, View } from 'react-native'
import { createStackNavigator } from 'react-navigation'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { tertiaryHeaderStyles } from '../components/layout/header-styles'
import {
  CustomText,
  Container,
  FooterActions,
  CustomHeader,
} from '../components'
import { eulaRoute, restoreRoute } from '../common'
import { eulaAccept } from './eula-store'
import type { Store } from '../store/type-store'
import { EULA_URL, localEulaSource } from './type-eula'
import { color } from '../common/styles/constant'
import { LoaderGif } from '../components/loader-gif/loader-gif'
import type { EulaScreenState } from './type-eula'
import type { CustomError } from '../common/type-common'

export class EulaScreen extends PureComponent<*, EulaScreenState> {
  state = {
    error: null,
  }

  static navigationOptions = ({ navigation }) => ({
    header: (
      <CustomHeader
        backgroundColor={color.bg.tertiary.color}
        flatHeader
        centerComponent={
          <CustomText bg="tertiary" tertiary transparentBg semiBold>
            Terms and Conditions
          </CustomText>
        }
      />
    ),
    swipeEnabled: false,
  })

  onReject = () => {
    Alert.alert(
      'Alert',
      'You will not be able to use the application without accepting Terms and Conditions'
    )
  }

  onError = (error: CustomError) => {
    if (error) {
      this.setState({ error })
    }
  }

  onAccept = () => {
    this.props.eulaAccept(true)
    this.props.navigation.navigate(restoreRoute)
  }

  render() {
    const webViewUri = this.state.error ? localEulaSource : EULA_URL

    return (
      <Container fifth>
        <WebView
          source={{ uri: webViewUri }}
          startInLoadingState={true}
          renderLoading={() => LoaderGif}
          onError={this.onError}
          renderError={() => <View />}
        />
        <FooterActions
          onAccept={this.onAccept}
          onDecline={this.onReject}
          denyTitle="Decline"
          acceptTitle="Accept"
          testID="eula"
        />
      </Container>
    )
  }
}

const mapStateToProps = ({ eula }: Store) => ({
  eula,
})

const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      eulaAccept,
    },
    dispatch
  )

export default createStackNavigator({
  [eulaRoute]: {
    screen: connect(mapStateToProps, mapDispatchToProps)(EulaScreen),
  },
})
