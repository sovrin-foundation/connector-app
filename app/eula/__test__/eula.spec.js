// @flow
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux'
import { EulaScreen } from '../eula'
import { getNavigation, getStore } from '../../../__mocks__/static-data'

describe('Eula screen', () => {
  const navigation = getNavigation()
  const store = getStore()

  it('should render properly and snapshot should match', () => {
    const eula = {
      isEulaAccept: false,
    }
    const tree = renderer
      .create(
        <Provider store={store}>
          <EulaScreen eula={eula} navigation={navigation} />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
