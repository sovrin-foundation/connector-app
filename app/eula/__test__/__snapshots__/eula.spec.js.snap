// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Eula screen should render properly and snapshot should match 1`] = `
<View
  fifth={true}
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <WebView
    onError={[Function]}
    renderError={[Function]}
    renderLoading={[Function]}
    source={
      Object {
        "uri": "https://www.connect.me/ios_eula.html",
      }
    }
    startInLoadingState={true}
  />
  <RCTSafeAreaView
    style={
      Object {
        "backgroundColor": "rgba(119, 119, 119, 1.0)",
      }
    }
  >
    <View
      fifth={true}
      style={
        Object {
          "backgroundColor": "#ffffff",
        }
      }
    >
      <View
        style={Object {}}
      >
        <View
          customColor={
            Object {
              "backgroundColor": "rgba(119, 119, 119, 1.0)",
            }
          }
          row={true}
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={Object {}}
            >
              <View
                style={
                  Array [
                    Object {
                      "marginLeft": 15,
                      "marginRight": 15,
                    },
                    undefined,
                    Object {
                      "marginLeft": 0,
                      "marginRight": 0,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel="eula-deny"
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      undefined,
                    ]
                  }
                  testID="eula-deny"
                  tvParallaxProperties={undefined}
                >
                  <View
                    pointerEvents="box-only"
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#9E9E9E",
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "padding": 19,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "backgroundColor": "rgba(119, 119, 119, 1.0)",
                        },
                        undefined,
                        Object {
                          "padding": 12,
                        },
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "paddingVertical": 17,
                          },
                        ],
                        false,
                        false,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {},
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 16,
                            },
                            Object {
                              "color": "#ffffff",
                            },
                            Object {
                              "fontSize": 14,
                            },
                            undefined,
                            undefined,
                            undefined,
                            undefined,
                            false,
                          ],
                        ]
                      }
                    >
                      Decline
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={Object {}}
            >
              <View
                style={
                  Array [
                    Object {
                      "marginLeft": 15,
                      "marginRight": 15,
                    },
                    undefined,
                    Object {
                      "marginLeft": 0,
                      "marginRight": 0,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel="eula-accept"
                  accessibilityTraits={undefined}
                  accessible={true}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                      },
                      Array [
                        Object {
                          "borderLeftColor": "#ffffff",
                          "borderLeftWidth": 0.4,
                        },
                      ],
                    ]
                  }
                  testID="eula-accept"
                  tvParallaxProperties={undefined}
                >
                  <View
                    pointerEvents="box-only"
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#9E9E9E",
                          "flexDirection": "row",
                          "justifyContent": "center",
                          "padding": 19,
                        },
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        undefined,
                        Object {
                          "backgroundColor": "rgba(119, 119, 119, 1.0)",
                        },
                        undefined,
                        Object {
                          "padding": 12,
                        },
                        undefined,
                        undefined,
                        undefined,
                        Array [
                          Object {
                            "paddingVertical": 17,
                          },
                          Object {
                            "borderLeftColor": "#ffffff",
                            "borderLeftWidth": 0.4,
                          },
                        ],
                        false,
                        false,
                      ]
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Array [
                          Object {},
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          Array [
                            Object {
                              "color": "white",
                              "fontSize": 16,
                            },
                            Object {
                              "color": "#ffffff",
                            },
                            Object {
                              "fontSize": 14,
                            },
                            undefined,
                            undefined,
                            Object {
                              "fontWeight": "bold",
                            },
                            undefined,
                            false,
                          ],
                        ]
                      }
                    >
                      Accept
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
